src/main/java/ch/njol/skript/expressions/ExprSaturation.java

```diff
@@ -62,45 +69,50 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	}
 	
 	@Override
-	protected Number[] get(final Event e, final Player[] source) {
-		return get(source, new Converter<Player, Number>() {
+	protected Number[] get(final Event event, final Player[] source) {
+		return get(source, new Getter<Float, Player>() {
 			@Override
-			@Nullable
-			public Number convert(final Player p) {
-				return p.getSaturation();
+			public Float get(final Player player) {
+				return player.getSaturation();
 			}
 		});
 	}
 	
-	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET)
-			return new Class[] {Number.class};
-		return super.acceptChange(mode);
-	}
-	
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		if (mode == ChangeMode.SET) {
-			assert delta != null;
-			for (final Player p : getExpr().getArray(e)) {
-				assert p != null : getExpr();
-				p.setSaturation(((Number) delta[0]).floatValue());
-			}
-		} else {
-			super.change(e, delta, mode);
-		}
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		return CollectionUtils.array(Number.class);
 	}
 	
+	@SuppressWarnings("null")
 	@Override
-	public Class<Number> getReturnType() {
-		return Number.class;
+	public void change(Event event, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		float value = ((Number)delta[0]).floatValue();
+		switch (mode) {
+			case ADD:
+				for (Player player : getExpr().getArray(event))
+					player.setSaturation(player.getSaturation() + value);
+				break;
+			case REMOVE:
+				for (Player player : getExpr().getArray(event))
+					player.setSaturation(player.getSaturation() - value);
+				break;
+			case SET:
+				for (Player player : getExpr().getArray(event))
+					player.setSaturation(value);
+				break;
+			case DELETE:
+			case REMOVE_ALL:
```

Comment:
Again, remove all != delete.

Reply:
```
clear saturation of player
remove all the saturation from the player
reset saturation of the player
```
?

What are these reviews

Reply:
Remove all doesn't work like that, with your example, it would be:
```
remove all saturation from saturation
```

Reply:
wth? That would be equivalent to `remove all 2.25 from 2.25` That will not work.

Reply:
Oh, you misunderstood `remove all`. It's meant to be used like : 
```
remove all "string repeated inside the variable" from {_list::*}
remove all stone from the player
```

Reply:
https://github.com/bensku/Skript/blob/master/src/main/java/ch/njol/skript/effects/EffChange.java#L86

Remove all isn't normally used in expression because of this.